trpcultivate_germplasm/tests/src/Kernel/TripalImporter/GermplasmAccessionImporter/GermplasmAccessionImporterTest.php

```diff
@@ -0,0 +1,668 @@
+    $non_existent_organism_id = $this->importer->getOrganismID('Nullus', 'organismus', '');
+    $printed_output = ob_get_clean();
+    $this->assertTrue($printed_output == 'Could not find an organism "Nullus organismus" in the database.', "Did not get the expected error message when testing for a non-existant organism.");
+
+    // Not testing if multiple organisms are retrieved, since Chado should be preventing such a situation
+  }
+
+  /**
+   * Tests focusing on the Germplasm Accession Importer getStockID() function
+   *
+   * @group germ_accession_importer
+   */
+  public function testGermplasmAccessionImporterGetStockID() {
+
+    // Insert an organism
+    $subtaxa_cvterm_id = $this->getCVtermID('TAXRANK', '0000023');
+
+    $organism_id = $this->connection->insert('1:organism')
+      ->fields([
+        'genus' => 'Tripalus',
+        'species' => 'databasica',
+        'infraspecific_name' => 'chadoii',
+        'type_id' => $subtaxa_cvterm_id,
+      ])
+      ->execute();
+
+    // Insert a stock
+    $stock_id = $this->connection->insert('1:stock')
+      ->fields([
+        'organism_id' => $organism_id,
+        'name' => 'stock1',
+        'uniquename' => 'TEST:1',
+        'type_id' => 9,
+      ])
+      ->execute();
+
+    // Test that the stock just inserted gets selected
+    $grabbed_stock_id = $this->importer->getStockID('stock1', 'TEST:1', $organism_id);
+    $this->assertEquals($grabbed_stock_id, $stock_id, "The stock ID grabbed by the importer does not match the one that was inserted into the database.");
+
+    // Test that a stock not in the database successfully gets inserted
+    ob_start();
+    $created_stock_id = $this->importer->getStockID('stock2', 'TEST:2', $organism_id);
+    $printed_output = ob_get_clean();
+    $this->assertTrue($printed_output == 'Inserting "stock2".', "Did not get the expected notice message when inserting a new stock.");
+
+    $stock2_query = $this->connection->select('1:stock', 's')
+      ->fields('s', ['stock_id'])
+      ->condition('organism_id', $organism_id, '=')
+      ->condition('name', 'stock2', '=')
+      ->condition('uniquename', 'TEST:2', '=')
+      ->condition('type_id', 9, '=');
+    $stock2_record = $stock2_query->execute()->fetchAll();
+    $this->assertEquals($created_stock_id, $stock2_record[0]->stock_id, "The stock ID inserted for \"stock2\" does not match the stock ID returned by getStockID().");
+
+    // No test for if the insert fails, since most likely will get a complaint from Chado
+
+    // Test for a stock name + organism that already exists but has a different accession
+    ob_start();
+    $grabbed_dup_stock_name = $this->importer->getStockID('stock1', 'TEST:1000', $organism_id);
+    $printed_output = ob_get_clean();
+    $this->assertTrue($printed_output == 'A stock already exists for "stock1" but with an accession of "TEST:1" which does not match the input file.', "Did not get the expected error message when testing for duplicate stock names.");
```

Comment:
It's always best to pass the things being compared to the assert directly rather then doing logic or pre-checking. For example, this is better done using assertEquals() because then when it fails you see both the expected and printed text.
```suggestion
    $this->assertEquals('A stock already exists for "stock1" but with an accession of "TEST:1" which does not match the input file.', $printed_output,
      "Did not get the expected error message when testing for duplicate stock names.");
```

Reply:
Resolved in 09f6fe7